frontend/src/libraries/explorer-wamp/transactions.ts

```diff
@@ -173,36 +172,22 @@ export default class TransactionsApi extends ExplorerApi {
       if (transactions.length > 0) {
         await Promise.all(
           transactions.map(async (transaction) => {
-            // TODO: Expose transaction status via transactions list from chunk
-            // RPC, and store it during Explorer synchronization.
-            //
-            // Meanwhile, we query this information in a non-effective manner,
-            // that is making a separate query per transaction to nearcore RPC.
-            const [transactionExtraInfo, finalTimestamp] = await Promise.all([
-              this.call<any>("nearcore-tx", [
-                transaction.hash,
-                transaction.signerId,
-              ]),
-              this.queryFinalTimestamp(),
+            const actions = await this.call<any>("select", [
+              `SELECT actions.transaction_hash, actions.action_index, actions.action_type as kind, actions.action_args as args
+            FROM actions
+            WHERE actions.transaction_hash = :hash
+            ORDER BY actions.action_index`,
+              {
+                hash: transaction.hash,
+              },
             ]);
-            transaction.status = Object.keys(
-              transactionExtraInfo.status
-            )[0] as ExecutionStatus;
-            transaction.isFinal = transaction.blockTimestamp <= finalTimestamp;
-            // Given we already queried the information from the node, we can use the actions,
-            // since DeployContract.code and FunctionCall.args are stripped away due to their size.
-            //
-            // Once the above TODO is resolved, we should just move this to TransactionInfo method
-            // (i.e. query the information there only for the specific transaction).
-            const actions = transactionExtraInfo.transaction.actions;
-            transaction.actions = actions.map((action: RpcAction | string) => {
+            transaction.actions = actions.map((action: any) => {
               if (typeof action === "string") {
                 return { kind: action, args: {} };
```

Comment:
This branch is not possible when we fetch the information from the DB since we already preprocessed the data.